pkg/metrics/metrics.go

```diff
@@ -3,12 +3,71 @@ package metrics
 import (
 	"errors"
 	"net/http"
+	"time"
 
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+type Metrics struct {
+	AgentInfo        *prometheus.GaugeVec
+	SigningRequests  *prometheus.HistogramVec
+	SSHRestartsCount prometheus.Counter
+}
+
+func NewMetrics() *Metrics {
+	m := &Metrics{
+		AgentInfo: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "agent_info",
+				Help: "Information about the agent version, SSH version and stack ID",
+			},
+			[]string{"version", "ssh_version", "stack_id"},
+		),
+		SigningRequests: newNativeHistogramVec(
+			prometheus.HistogramOpts{
+				Name:    "signing_requests_duration_seconds",
+				Help:    "Duration of signing requests in seconds",
+				Buckets: []float64{0.5, 1, 2, 5, 10}, // TODO: redefine boundaries if needed after understanding the distribution
```

Comment:
As we're using native histograms, IIUC we don't really need to define our own buckets.